# What is the expected earning of a student who picks a Colorado University at random? If they do not graduate, they will spontaneously combust and as such will not make any earnings.

The random variable I am studying is the earnings of a student who picks a Colorado University at random. One trial includes picking a university at a fair random, then deciding if they graduate based on that college's graduation rate. The outcome I measuring the average earnings of students.

In [1]:
import pandas, seaborn, random

In [3]:
college_data = [ #Name, Cost, Grad Rate, Median Earnings
    ["Colorado School of Mines", 24980, 0.85, 90060],
    ["University of Colorado Denver", 13594, 0.48, 64107],
    ["University of Colorado Boulder", 21849, 0.68, 56900]
]

In [9]:
numSims = 1000000
tot = 0

for _ in range(numSims):
    college = random.choice(college_data)

    options = ["Grad", "Not"]
    weights = [college[2], 1-college[2]]

    if random.choices(options, weights = weights, k =1)[0] == "Grad":
        tot += college[3]

print(tot/numSims)

48724.122504


Theoretical Earnings: $48,671.45

The expected earnings of a student who picks a Colorado University at random empirically are 48,724.12 dollars based on 1,000,000 simulations. Theoretically, the earnings are 48,641.45 dollars.